pkg/engine/mutate/strategicPreprocessing_test.go

```diff
@@ -125,6 +125,130 @@ func Test_preProcessStrategicMergePatch_AnnotationMap(t *testing.T) {
 	}
 }
 
+func Test_preProcessStrategicMergePatch_DifferentKinds(t *testing.T) {
+	rawPolicy := []byte(`{
+		"spec": {
+		  "imagePullSecrets": [
+			{
+			  "name": "regcred"
+			}
+		  ],
+		  "initContainers": [
+			{
+			  "(image)": "index.docker.io/*"
+			}
+		  ]
+		}
+	  }`)
+
+	rawResource := []byte(`{
+		"spec": [
+			"containers": {
+				"{\"name\":\"nginx\"}": {
+				  "image": {},
+				  "imagePullPolicy": {},
+				  "name": {},
+				  "resources": {}
+				}
+			}
+		]
+	}`)
+
+	expected := `{"spec": {"imagePullSecrets": [{"name": "regcred"}], "initContainers": [{}]}}`
```

Comment:
Why this policy applies? There's no initContiainer defined in the resource, so the condition does not match here. Shouldn't we return an empty pattern?

Reply:
This is not applied also. Because k8s patch is applied only if patch element has `name` field. It uses `name` for navigation. In our case we have empty initContainers. So it should not be applied. I will check on live cluster.